op-node/rollup/derive/channel_out.go

```diff
@@ -145,24 +250,36 @@ func (co *ChannelOut) InputBytes() int {
 // Use `Flush` or `Close` to move data from the compression buffer into the ready buffer if more bytes
 // are needed. Add blocks may add to the ready buffer, but it is not guaranteed due to the compression stage.
 func (co *ChannelOut) ReadyBytes() int {
-	return co.compress.Len()
+	return co.reader.Len()
 }
 
 // Flush flushes the internal compression stage to the ready buffer. It enables pulling a larger & more
 // complete frame. It reduces the compression efficiency.
 func (co *ChannelOut) Flush() error {
-	return co.compress.Flush()
+	if err := co.compress.Flush(); err != nil {
+		return err
+	}
+	if co.batchType == SpanBatchType && co.closed && co.reader.Len() == 0 && co.compress.Len() > 0 {
```

Comment:
```suggestion
	if co.batchType == SpanBatchType && co.closed && co.ReadyBytes() == 0 && co.compress.Len() > 0 {
```

Also can we apply `co.ReadyBytes()`s to other implementations?